go/test/endtoend/shardedrecovery/sharded_recovery_test.go

```diff
@@ -0,0 +1,630 @@
+	assert.Nil(t, err)
+	insertData(t, master, 1)
+	checkData(t, replica1, 1)
+
+	// insert more data on the master
+	insertData(t, master, 2)
+
+	// backup the replica
+	err = localCluster.VtctlclientProcess.ExecuteCommand("Backup", replica1.Alias)
+	assert.Nil(t, err)
+
+	// check that the backup shows up in the listing
+	output, err := localCluster.ListBackups("test_keyspace/0")
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(output))
+	assert.True(t, strings.HasSuffix(output[0], replica1.Alias))
+
+	// insert more data on the master
+	insertData(t, master, 3)
+
+	err = localCluster.VtctlclientProcess.ApplyVSchema(keyspaceName, vSchema)
+	assert.Nil(t, err)
+
+	// create the split shards
+	for _, tablet := range []*cluster.Vttablet{shard0Master, shard0Replica, shard0RdOnly} {
+		err = localCluster.VtctlclientProcess.InitTablet(tablet, cell, keyspaceName, hostname, shard0Name)
+		assert.Nil(t, err)
+	}
+
+	for _, tablet := range []*cluster.Vttablet{shard1Master, shard1Replica, shard1RdOnly} {
+		err = localCluster.VtctlclientProcess.InitTablet(tablet, cell, keyspaceName, hostname, shard1Name)
+		assert.Nil(t, err)
+	}
+
+	for _, tablet := range []*cluster.Vttablet{shard0Master, shard0Replica, shard0RdOnly, shard1Master, shard1Replica, shard1RdOnly} {
+		tablet.VttabletProcess.ExtraArgs = []string{"-binlog_use_v3_resharding_mode=true"}
+		err = tablet.VttabletProcess.Setup()
+		assert.Nil(t, err)
+	}
+
+	err = localCluster.VtctlProcess.ExecuteCommand("InitShardMaster", "-force", "test_keyspace/-80", shard0Master.Alias)
+	assert.Nil(t, err)
+
+	err = localCluster.VtctlProcess.ExecuteCommand("InitShardMaster", "-force", "test_keyspace/80-", shard1Master.Alias)
+	assert.Nil(t, err)
+
+	shardedTablets := []*cluster.Vttablet{shard0Master, shard0Replica, shard0RdOnly, shard1Master, shard1Replica, shard1RdOnly}
+
+	for _, tablet := range shardedTablets {
+		_ = tablet.VttabletProcess.WaitForTabletType("SERVING")
+		assert.Nil(t, err)
+	}
+
+	for _, tablet := range shardedTablets {
+		assert.Equal(t, tablet.VttabletProcess.GetTabletStatus(), "SERVING")
+	}
+
+	// we need to create the schema, and the worker will do data copying
+	for _, keyspaceShard := range []string{"test_keyspace/-80", "test_keyspace/80-"} {
+		err = localCluster.VtctlclientProcess.ExecuteCommand("CopySchemaShard", "test_keyspace/0", keyspaceShard)
+		assert.Nil(t, err)
+	}
+
+	err = localCluster.VtctlclientProcess.ExecuteCommand("SplitClone", "test_keyspace", "0", "-80,80-")
+	assert.Nil(t, err)
+
+	err = localCluster.VtctlclientProcess.ExecuteCommand("MigrateServedTypes", "test_keyspace/0", "rdonly")
+	assert.Nil(t, err)
+
+	err = localCluster.VtctlclientProcess.ExecuteCommand("MigrateServedTypes", "test_keyspace/0", "replica")
+	assert.Nil(t, err)
+
+	// then serve master from the split shards
+	err = localCluster.VtctlclientProcess.ExecuteCommand("MigrateServedTypes", "test_keyspace/0", "master")
+	assert.Nil(t, err)
```

Comment:
@sougou should we change this to use the new `Reshard`, `MigrateReads` and `MigrateWrites`?

Reply:
We should write separate tests for the new workflows. @rohit-nayak-ps is taking this on.